WormholeConnect-Migration-Guide-v2.5.0-to-v3.0.0.md

```diff
@@ -0,0 +1,227 @@
+# WormholeConnect Migration Guide: v2.5.0 to v3.0.0
+
+This guide covers the migration from WormholeConnect v2.5.0 to v3.0.0, focusing on the public interface changes that affect integrators.
+
+## Summary
+
+**Good news!** The core `<WormholeConnect />` component interface remains **fully backward compatible** between v2.5.0 and v3.0.0. Most integrators can upgrade without any code changes.
+
+## Component Interface (No Changes Required)
+
+The main component props remain identical:
+
+```typescript
+// ✅ Same in both v2.5.0 and v3.0.0
+<WormholeConnect 
+  theme={theme}    // Optional - WormholeConnectTheme
+  config={config}  // Optional - WormholeConnectConfig
+/>
+```
+
+### Theme Interface (No Changes)
+
+```typescript
+// ✅ Unchanged - works exactly the same
+const theme: WormholeConnectTheme = {
+  mode: 'light', // or 'dark'
+  primary: '#ff6b35',
+  secondary: '#1976d2',
+  // ... all other theme properties work the same
+};
+```
+
+## Configuration Enhancements (Optional)
+
+While the core interface is unchanged, v3.0.0 adds **optional** new configuration capabilities:
+
+### 1. Enhanced Token Filtering
+
+```typescript
+// ✅ v2.5.0 - Still works in v3.0.0
+const config: WormholeConnectConfig = {
+  isTokenSupportedHandler: (token) => {
+    return token.symbol !== 'BLOCKED_TOKEN';
+  }
+};
+
+// 🆕 v3.0.0 - New optional sourceToken parameter
+const config: WormholeConnectConfig = {
+  isTokenSupportedHandler: (token, sourceToken) => {
+    // Can now filter destination tokens based on source token
+    if (sourceToken?.symbol === 'USDC' && token.symbol === 'BANNED_PAIR') {
+      return false;
+    }
+    return token.symbol !== 'BLOCKED_TOKEN';
+  }
+};
+```
+
+### 2. Route Filtering (New Feature)
+
+```typescript
+// 🆕 v3.0.0 - Optional route filtering
+const config: WormholeConnectConfig = {
+  filterRoutes: (routes) => {
+    // Hide certain routes from users
+    return routes.filter(route => route !== 'TokenBridge');
+  }
+};
+```
+
+## Import Changes
+
+### Main Import (Recommended - No Changes)
+
+```typescript
+// ✅ Works the same in both versions
+import WormholeConnect from '@wormhole-foundation/wormhole-connect';
+```
+
+### Detailed Imports (Minor Changes)
+
+If you import specific types or utilities:
+
+```typescript
+// ✅ v2.5.0
+import WormholeConnect, { 
+  WormholeConnectConfig,
+  WormholeConnectTheme,
+  MAINNET,
+  TESTNET 
+} from '@wormhole-foundation/wormhole-connect';
+
+// ✅ v3.0.0 - WormholeConnectConfig moved to config namespace
+import WormholeConnect, { 
+  config,              // 🆕 Config types now in namespace
```

Comment:
This is correct, but our main documentation is outdated now:
https://wormhole.com/docs/products/connect/get-started/